plugins/generator-1.19.4/forge-1.19.4/templates/structure/structure.java.ftl

```diff
@@ -0,0 +1,152 @@
+ # (at your option) any later version.
+ #
+ # This program is distributed in the hope that it will be useful,
+ # but WITHOUT ANY WARRANTY; without even the implied warranty of
+ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ # GNU General Public License for more details.
+ #
+ # You should have received a copy of the GNU General Public License
+ # along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ #
+ # Additional permission for code generator templates (*.ftl files)
+ #
+ # As a special exception, you may create a larger work that contains part or
+ # all of the MCreator code generator templates (*.ftl files) and distribute
+ # that work under terms of your choice, so long as that work isn't itself a
+ # template for code generation. Alternatively, if you modify or redistribute
+ # the template itself, you may (at your option) remove this special exception,
+ # which will cause the template and the resulting code generator output files
+ # to be licensed under the GNU General Public License without this special
+ # exception.
+-->
+
+<#-- @formatter:off -->
+<#include "../mcitems.ftl">
+<#include "../procedures.java.ftl">
+
+package ${package}.world.features;
+
+<#compress>
+public class ${name}Feature extends Feature<NoneFeatureConfiguration> {
+
+	public static ${name}Feature FEATURE = null;
+
+	public static Feature<?> feature() {
+		FEATURE = new ${name}Feature();
+		return FEATURE;
+	}
+
+	private final Set<ResourceKey<Level>> generate_dimensions = Set.of(
+		<#list data.spawnWorldTypes as worldType>
+			<#if worldType == "Surface">
+				Level.OVERWORLD
+			<#elseif worldType == "Nether">
+				Level.NETHER
+			<#elseif worldType == "End">
+				Level.END
+			<#else>
+				ResourceKey.create(Registry.DIMENSION_REGISTRY,
+					new ResourceLocation("${generator.getResourceLocationForModElement(worldType.toString().replace("CUSTOM:", ""))}"))
+			</#if><#sep>,
+		</#list>
+	);
+
+	<#if data.restrictionBlocks?has_content>
+	private final List<Block> base_blocks;
+	</#if>
+
+	private StructureTemplate template = null;
+
+	public ${name}Feature() {
+		super(NoneFeatureConfiguration.CODEC);
+
+		<#if data.restrictionBlocks?has_content>
+			base_blocks = List.of(
+				<#list data.restrictionBlocks as restrictionBlock>
+					${mappedBlockToBlock(restrictionBlock)}<#sep>,
+				</#list>
+			);
+		</#if>
+	}
+
+	@Override public boolean place(FeaturePlaceContext<NoneFeatureConfiguration> context) {
+		if (!generate_dimensions.contains(context.level().getLevel().dimension()))
+			return false;
+
+		if (template == null)
+			template = context.level().getLevel().getStructureManager()
+					.getOrCreate(new ResourceLocation("${modid}", "${data.structure}"));
+
+		if (template == null)
+			return false;
+
+		boolean anyPlaced = false;
+		if ((context.random().nextInt(1000000) + 1) <= ${data.spawnProbability}) {
+			int count = context.random().nextInt(
+			<#if ((data.maxCountPerChunk) > (data.minCountPerChunk))>

```

Comment:
We have validation for this in UI now so this "hack" should be reverted to how it was before

Reply:
oh, this was written when I was porting to 1.19.3 and it was still an issue, good to know